bridge/core/api/form_data.cc

```diff
@@ -0,0 +1,193 @@
+    return nullptr;
+  }
+
+  // 查找数据
+  for (const auto& part : _parts) {
+    if (part->GetName() == name.ToStdString(ctx())) {
+      return &*part->getFirst();
+    }
+  }
+
+  // 如果没有找到，则返回 nullptr
+  return nullptr;
+}
+
+std::vector<BlobPart::ImplType> FormData::getAll(const AtomicString& name, ExceptionState& exception_state) {
+  std::vector<BlobPart::ImplType> result;
+
+  // 验证参数有效性
+  if (name.IsEmpty()) {
+    exception_state.ThrowException(ctx(), ErrorType::ArgumentError, "The name parameter must not be empty.");
+    return result;
+  }
+
+  // 收集数据
+  for (const auto& part : _parts) {
+    if (part->GetName() == name.ToStdString(ctx())) {
+      for (const auto& value : part->GetValues()) {
+        result.push_back(std::make_shared<BlobPart>(value));
+      }
+    }
+  }
+
+  return result;
+}
+
+bool FormData::has(const AtomicString& name, ExceptionState& exception_state) {
+  // 验证参数有效性
+  if (name.IsEmpty()) {
+    exception_state.ThrowException(ctx(), ErrorType::ArgumentError, "The name parameter must not be empty.");
+    return false;
+  }
+
+  // 检查数据
+  for (const auto& part : _parts) {
+    if (part->GetName() == name.ToStdString(ctx())) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void FormData::set(const AtomicString& name,
+                   const std::shared_ptr<webf::BlobPart>& value,
+                   const AtomicString& fileName,
+                   ExceptionState& exception_state) {
+  // 验证参数有效性
+  if (name.IsEmpty()) {
+    exception_state.ThrowException(ctx(), ErrorType::ArgumentError, "The name parameter must not be empty.");
+    return;
+  }
+
+  // 移除已存在的相同名称的条目
+  _parts.erase(std::remove_if(_parts.begin(), _parts.end(),
+                              [name, this](const std::shared_ptr<FormDataPart>& part) {
+                                return part->GetName() == name.ToStdString(ctx());
+                              }),
+               _parts.end());
+
+  // 创建 FormDataPart 对象
+  auto form_data_part = std::make_shared<FormDataPart>(name.ToStdString(ctx()), value, fileName.ToStdString(ctx()));
+
+  // 添加新的数据
+  _parts.push_back(form_data_part);
+}
+
+void FormData::forEach(const std::shared_ptr<QJSFunction>& callback,
+                       const ScriptValue& thisArg,
+                       ExceptionState& exception_state) {
+  if (!callback || !callback->IsFunction(ctx())) {
+    exception_state.ThrowException(ctx(), ErrorType::ArgumentError, "The callback function must be callable.");
+    return;
+  }
+  // callbackFn:(value: BlobPart, key: string, parent: FormData) => void
+  for (const auto& part : _parts) {
+    ScriptValue args[3];
+    /*value*/ args[0] = ScriptValue(ctx(), part->ToQuickJS(ctx()));
+    /*key*/ args[1] = ScriptValue(ctx(), AtomicString(ctx(), part->GetName()));
+
+    // TODO: which parent???
+    /*parent*/ args[2] = ScriptValue(ctx(), this->ToQuickJS());
+
+    // 调用回调函数
```

Comment:
annotations should be written in English

Reply:
Chinese annotations removed.